src/ott/solvers/nn/losses.py

```diff
@@ -0,0 +1,85 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import jax
+import jax.numpy as jnp
+from ott.geometry import pointcloud
+from ott.solvers.linear import sinkhorn
+from ott.problems.linear import linear_problem
+import ott.geometry.costs as costs
+from typing import Any, Mapping, Callable
+from types import MappingProxyType
+
+class MongeGap:
```

Comment:
In general, I think having a class for this too complex. Can you please refactor this as 
a function? Here's a suggestion:
```python
def monge_gap(
    x: jnp.ndarray, y: jnp.ndarray,
    cost_fn: Optional[costs.CostFn] = None,
    epsilon: Optional[float] = None,
    relative_epsilno: Optional[bool] = None,
    scale_cost: ... = 1.0,
    **kwargs: Any
) -> float:
    ...
```
Alternatively, we can have `geom_kwargs: Mapping[str, Any], solver_kwargs: Mapping[str, Any]` as before instead of exposing `PointCloud` arguments, no strong preference.